app/js/components/pages/Job.jsx

```diff
@@ -73,13 +74,23 @@ const Job = ({ plugin, workflow, inputArtifacts, onClickSubmit, onClickCancel })
             </form>
             <button
                 className="btn btn-danger"
-                onClick={onClickCancel}
+                onClick={cancelJob}
             >
                 Cancel
             </button>
             <button
                 className="btn btn-primary pull-right"
-                onClick={onClickSubmit}
+                onClick={() => {
+                    const formData = new FormData(document.querySelector('form'));
+                    for (const [key, value] of formData.entries()) {
+                        // this could eventually decorate the inputs that are invalid
```

Comment:
Ideally the input forms will understand the predicates outright and will actively prevent bad input.